onward/app/feed/DeeplyReadAgent.scala

```diff
@@ -123,10 +121,13 @@ class DeeplyReadAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi) ex
       // We now perform the atomic update of deeplyReadItems to faithfully reflects the state of the Ophan answer.
       // It is done as last step of the process because by then the CAPI content has been loaded.
       }
-    fDeeplyReadItemsWithCapi.foreach { sequence =>
-      val mapDeeplyReadItems = sequence.filter(_.isDefined).map(_.get).toMap
-      deeplyReadItems = collection.mutable.Map(mapDeeplyReadItems.toSeq: _*)
-    }
+    fDeeplyReadItemsWithCapi.onComplete({
+      case Success(sequence) =>
+        val mapDeeplyReadItems = sequence.filter(_.isDefined).map(_.get).toMap
+        deeplyReadItems = collection.mutable.Map(mapDeeplyReadItems.toSeq: _*)
+      case Failure(exception) =>
```

Comment:
In case of failure will it retry or does it just fire an error?

Reply:
Just logs the error message at the moment, if we discover what the issue is and decide that retries are the best way to solve it then we could look into that.

Reply:
👍